src/object/cli_obj.c

```diff
@@ -1109,7 +1110,7 @@ obj_shard_tgts_query(struct dc_object *obj, uint32_t map_ver, uint32_t shard,
 	shard_tgt->st_ec_tgt = ec_tgt_idx;
 	start_shard = shard - ec_tgt_idx;
 
-	if (obj_auxi->is_ec_obj &&
+	if (obj_auxi->is_ec_obj && !obj_auxi->spec_shard &&
 	    (obj_auxi->csum_retry || obj_auxi->tx_uncertain ||
 	     obj_auxi->force_degraded || DAOS_FAIL_CHECK(DAOS_OBJ_FORCE_DEGRADE))) {
```

Comment:
this is only for the case of EC obj csum_retry/tx_uncertain/force_degraded, will it affect EC agg's dsc_obj_fetch(spec_shard) case? seems L1113's change is not must?

Reply:
yes, this change will not affect EC agg.  But for spec shard case, it should not try ec degraded fetch, so I added this check.